lua/indent_blankline/init.lua

```diff
@@ -125,32 +117,44 @@ local refresh = function()
     local char_highlight_list = v("indent_blankline_char_highlight_list")
     local space_char_highlight_list = v("indent_blankline_space_char_highlight_list")
     local space_char_blankline_highlight_list = v("indent_blankline_space_char_blankline_highlight_list")
-    local space_char = v("indent_blankline_space_char")
     local space_char_blankline = v("indent_blankline_space_char_blankline")
+
+    local space_character = vim.opt.listchars:get().space or " "
+    -- source tab chars in a way that works with UTF-8 characters
+    local tab_characters = vim.fn.split(vim.opt.listchars:get().tab, "\\zs")
```

Comment:
If there is no tab set, this resolves to `{ "n", "u", "l", "l" }` :smile: 
we should default to `{space_character}` here

Reply:
The issue here is that without a tab character set in `listchars` this doesn't really work anyway since Vim will display a literal `^I` with a fixed width of 2 columns, regardless of your tabstop setting.

![image](https://user-images.githubusercontent.com/72616153/131701862-724a17fd-dfbb-4c06-b6b5-a163a5526df7.png)
There is no handling for irregular indentation, so with `tabstop=4`, or anything over 2 really, the virtual text will be overlaid over actual line content.
![image](https://user-images.githubusercontent.com/72616153/131702683-a1a79dad-227c-4862-a071-e9842ae361c5.png)


The only scenario where this works out is when you set `tabstop=2`. I guess that is something that the user is probably aware of when they make the *active choice to remove default `listchars` without replacement*.  I wouldn't spent any energy trying to work out how to properly support this for other tabstop/shiftwidth settings.